tirtha_bk/tirtha/views.py

```diff
@@ -231,25 +257,56 @@ def _signin(token, create=False):
 
 
 @require_GET
-def googleAuth(request):
-    """
-    Sets auth token cookie post Google Auth.
+def google_login(request):
+
+    # build a full authorize callback uri
 
+    new_state = str(uuid.uuid4())  # Generate a new UUID
+    redirect_uri = request.build_absolute_uri('/signin-google') #TODO: build the redirect uri as per configugered in the GOOGLE BACKEND
+    request.session['auth_state'] = new_state
+
+    return oauth.google.authorize_redirect(request, redirect_uri, state = new_state)
+
+@require_GET
+def logout(request):
+    #user can signout after signing in if the user wants to
+    # build a logout button
     """
-    token = request.GET["token"]
-    output, contrib = _signin(token, create=True)
+    Pops the user from the session post logout.
+    The user is redirected to the index page.
+    """
+    request.session.pop('auth_token', None)
+    return redirect('index')
+
 
-    context = {"output": output, "banned": False, "blur": False}
+@require_GET
+def tokenAuth(request):
+    """
+    Sets auth token cookie post token-based authentication.
+    """                                
+    # stored_state = request.session.get('auth_state')
```

Comment:
@JeS24 Check if this is needed. @RudrasSingh Ignore.

Reply:
I think this won't be needed, we used it for just the debugging process. Still you can check it once!